src/main/java/sirius/db/es/Elastic.java

```diff
@@ -705,34 +705,25 @@ public <E extends ElasticEntity> ElasticQuery<E> select(Class<E> type) {
     /**
      * Creates a new query which can be used to fetch entities of the given types.
      * <p>
-     * Note: All entities to query across must extend a common super class.
+     * Note: All entities to query across must extend a common super class. Also note, that the list of types can be
+     * left empty and specified later via {@link ElasticQuery#withEffectiveIndices(List)}.
      *
      * @param commonSuperClass the common super class of all entities to query across
      * @param types            the types of the entities to query across
      * @param <E>              the generic common type of the entities to query across
      * @return a new query which can be used to fetch entities of the given types
      */
     @SafeVarargs
+    @SuppressWarnings("java:S1172")
+    @Explain("We only need to type for the compiler to enforce type safety.")
     public final <E extends ElasticEntity> ElasticQuery<E> selectMultiple(Class<E> commonSuperClass,
                                                                           Class<? extends E>... types) {
-        return new ElasticQuery<E>(mixing.getDescriptor(types[0]),
-                                   getLowLevelClient()).withAdditionalIndices(Arrays.stream(types).skip(1));
-    }
+        ElasticQuery<E> query = new ElasticQuery<>(null, getLowLevelClient());
+        if (types.length > 0) {
+            query.withEffectiveIndices(Arrays.stream(types).toList());
+        }
 
-    /**
-     * Creates a new query which can be used to fetch entities of the given types.
-     * <p>
-     * Note: All entities to query across must extend a common super class.
-     *
-     * @param commonSuperClass the common super class of all entities to query across
-     * @param types            the types of the entities to query across
-     * @param <E>              the generic common type of the entities to query across
-     * @return a new query which can be used to fetch entities of the given types
-     */
-    public final <E extends ElasticEntity> ElasticQuery<E> selectMultiple(Class<E> commonSuperClass,
-                                                                          List<Class<? extends E>> types) {
-        return new ElasticQuery<E>(mixing.getDescriptor(types.get(0)),
-                                   getLowLevelClient()).withAdditionalIndices(types.stream().skip(1));
```

Comment:
Why? I actually hate it when APIs only allow for varargs, as one has to always convert collections to arrays to use them.

---

This is also breaking

Reply:
Because selectMultiple(X).withEffectiveIndices(list) is already there..